# Data Preprocessing 
v0.0

#### Import the required libraries

In [1]:
import pandas as pd
import numpy as np

## Load the dataset

In [2]:
data_file = "./Datasets/all_price_data.csv"
dataset = pd.read_csv(data_file)

#### Display the first few rows of the dataset to check dataset

In [3]:
print(dataset.head())

   year  month     location  wholesale_price  retail_price
0  2017      1  Keppetipola            45.42          98.0
1  2017      1   Kurunegala              NaN           NaN
2  2017      1       Ampara              NaN         124.8
3  2017      1       Matale              NaN           NaN
4  2017      1     Vavuniya              NaN          78.0


## Handle Missing Values
### Count the number of missing values in each column

In [4]:
print(dataset.isnull().sum())

year                  0
month                 0
location              0
wholesale_price    1652
retail_price        791
dtype: int64


### Check how many records will be dropped if remove rows where both "wholesale_price" and "retail_price" have missing values

#### Count the number of records before dropping

In [5]:
num_records_before = len(dataset)
num_records_before

2664

#### Drop rows where both 'wholesale_price' and 'retail_price' are missing

In [6]:
dataset_cleaned = dataset.dropna(subset=['wholesale_price', 'retail_price'], how='all')

#### Count the number of records after dropping

In [7]:
num_records_after = len(dataset_cleaned)
num_records_after

1879

#### Calculate how many records were dropped

In [8]:
num_records_dropped = num_records_before - num_records_after

# Display the number of records dropped
print("Number of records dropped:", num_records_dropped)

Number of records dropped: 785


#### Check the number of missing values after cleaning

In [9]:
print(dataset_cleaned.isnull().sum())

year                 0
month                0
location             0
wholesale_price    867
retail_price         6
dtype: int64


- While employing this method, it's worth noting that numerous data records may be removed. To address this issue, we must handle missing values by replacing them with suitable, accurate values. 
- Despite these efforts, there are still missing values in the 'wholesale_price' column (867 instances) and the 'retail_price' column (6 instances). Therefore, it's essential to fill these missing values.

---------------------